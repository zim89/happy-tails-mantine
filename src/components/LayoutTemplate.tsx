import Footer from '@/modules/Footer';
import Header from '@/modules/Header';

export default function Layout({ children }: { children: React.ReactNode }) {
  return (
    <>
      <Header />
      {/* Padding top equals size of top navbar */}
      <div className='grid h-screen grid-rows-[_1fr_auto] pt-[74px] md:pt-[128px]'>
        <main>{children}</main>
        <Footer />
      </div>
    </>
  );
}
