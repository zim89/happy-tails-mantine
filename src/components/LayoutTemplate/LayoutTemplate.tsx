import CookiesNotification from '@/modules/CookiesNotification';
import Footer from '@/modules/Footer';
import Header from '@/modules/Header';

export default function Layout({ children }: { children: React.ReactNode }) {
  return (
    <>
      <Header />
      {/* Padding top equals size of top navbar */}
      <div className='grid h-screen grid-rows-[_1fr_auto] pt-[74px] lg:pt-[128px]'>
        <main>{children}</main>
        <CookiesNotification />
        <Footer />
      </div>
    </>
  );
}
