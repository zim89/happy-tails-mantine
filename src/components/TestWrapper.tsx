import { AppProviders } from '@/shared/config/AppProviders';
import theme from '@/shared/config/theme';
import { MantineProvider } from '@mantine/core';

type Props = {
  children: React.ReactNode;
};
export const TestWrapper = ({ children }: Props) => {
  return (
    <AppProviders>
      <MantineProvider theme={theme}>{children}</MantineProvider>
    </AppProviders>
  );
};
