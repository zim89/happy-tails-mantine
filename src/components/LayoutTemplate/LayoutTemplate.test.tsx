import { describe, test } from 'vitest';
import { render } from '@testing-library/react';

import { TestWrapper } from '../TestWrapper';
import LayoutTemplate from './';

const TestLayoutTemplate = ({ children }: { children: React.ReactNode }) => {
  return (
    <TestWrapper>
      <LayoutTemplate>{children}</LayoutTemplate>
    </TestWrapper>
  );
};

describe('LayoutTemplate', () => {
  test('renders correctly', () => {
    const { getByTestId } = render(
      <TestLayoutTemplate>
        <span data-testid='test'>Test</span>
      </TestLayoutTemplate>
    );

    const selection = getByTestId('test');

    expect(selection).toBeInTheDocument();
    expect(selection).toHaveTextContent('Test');
  });
});
