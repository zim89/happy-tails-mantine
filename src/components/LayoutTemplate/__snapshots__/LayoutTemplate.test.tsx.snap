// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LayoutTemplate > renders correctly 1`] = `
<span
  data-testid="test"
>
  Test
</span>
`;
